The Deceiver's Gambit: add TDG to mainline, replacing DM

Dalas121 opened the pull request:
Around two years ago Yumi asked me to create TDG as a replacement for Delfador's Memoirs, and gave me a rough script - the campaign covers Delfador's youth and Eldred's betrayal of Garard. Its gameplay includes a spellcasting mechanic for Delfador.

After this, I don't plan to drastically alter canon (like #9682) nor outright replace any campaigns (like this PR); TSG and TDG are a bit of a holdover from Yumi. DM has redeeming qualities, but it's a poor campaign - it was rated lower than every other mainline campaign in Hejnewar's recent community survey.

I plan to make some small tweaks to Liberty and HttT as well, to maintain continuity with TDG. For example, in L Maddock mentions "_the majority of my forces are still returning from the battle of Abez_" - but in TDG he refuses to send troops in the first place. That'll be a different PR sometime later.

AnekronCZ commented:
Glad to see TDG going in. I think it finally does justice to the character of Delfador and improves upon DM in basically every regard while maintaining a tighter connection with other mainline campaigns.

Wedge009 commented:
It's been a while since I played DM but I don't recall it being _that_ bad. Then again, I'm pretty easy to please - I also liked the demoted AOI campaign.

Whereas AOI was removed altogether, does TDG 'improve upon' DM in a similar way that your EI revision did or is a significant departure in terms of characterisation, story-telling, etc?

Dalas121 commented:
> Whereas AOI was removed altogether, does TDG 'improve upon' DM in a similar way that your EI revision did or is a significant departure in terms of characterisation, story-telling, etc?

I'm afraid TDG is pretty much a complete retcon and replacement. Figures like Methor, Leollyn, Lionel, Kalenz, or Iliah-Malal still exist, but the story is drastically different.

Wedge009 commented:
All right, I'll just have to give it a try some time.

Dalas121 commented:
> In the story portraits, is there some meaning behind the -wp suffix of the file names?

"WP" stands for "waterpixel" - that's the filter I ran to turn them from photos into pseudo-artwork. It's not important to keep them named -wp though.

> I skimmed over the text marked for translation and for the most part it looks good but I did find a few things. I list them here rather than noting them in the files directly because someone previously complained about notification 'spam'.

Great, thank you! I've fixed most of the issues you noticed (haven't pushed the commit yet). A couple of comments:
 
> S6 line 793+816: Capitalisation of King seems inconsistent throughout the campaign, but I highlight these because in between Deoran refers to a capitalised Queen.

Yeah... I had trouble staying consistent on this in EI too. I've fixed this here at least.

> Shouldn't Delfador invoke 'order against chaos'?

I'd meant "chaos against order" to mean that "order" is following the commands of the necromancer, while "chaos" is breaking free and doing your own thing.

> I haven't play-tested this, but is there no epilogue? It seems to end rather abruptly.

I wrote a couple draft epilogues, and went back and forth on this. Ultimately, I'm not sure exactly what happens after TDG ends, and I'm especially not sure how to make it satisfying for the player ("_good job killing Eldred, but Asheviere wins anyway_" isn't very exciting).

Asheviere's Dogs picks up immediately after TDG, with Asheviere recruiting orcs to put down a large-scale rebellion. Hopefully players feel that AD helps tie up TDG's ending - if not, I'll take another shot at an epilogue.

Wedge009 commented:
> "WP" stands for "waterpixel" - that's the filter I ran to turn them from photos into pseudo-artwork. It's not important to keep them named -wp though.

No worries, I was just curious.

> I'd meant "chaos against order" to mean that "order" is following the commands of the necromancer, while "chaos" is breaking free and doing your own thing.

I wondered if that's what you were intending. In my Christian thinking, I instinctively read 'order' as 'good' and 'chaos' as 'evil', since God - unambiguously good - is described in numerous places as bringing order to a chaotic/disordered world/universe. But even notwithstanding that, Wesnoth itself has 'lawful' and 'chaotic' alignments with the latter usually being associated with 'bad' and that's the context we have here with the necromancer, even though there are campaigns with chaotic factions as the protagonist (and Delfador is neutral?). And then there's D&D which treats the two spectrums as separate dimensions anyway.

I'm not sure if there's a better way of describing what you want here while keeping the poetic tone. I suppose it's okay for you to leave it as-is... but how about 'I invoke liberty against servitude!'? Or is that too explicit (in terms of description, not vulgarity)?

> I wrote a couple draft epilogues, and went back and forth on this. Ultimately, I'm not sure exactly what happens after TDG ends, and I'm especially not sure how to make it satisfying for the player ("_good job killing Eldred, but Asheviere wins anyway_" isn't very exciting).

As the author you're not sure? Or you mean in terms of what you inherited from nemaara?

Even without knowing the context of AD, it was pretty clear to me the last act is a set-up for HttT - it pretty much tells the story of the prologue, aside from Delfador's flight with Konrad. If the campaign is played on its own, then yes, the ending could seem pretty unsatisfying. But as often with the second part of a trilogy (eg Empire Strikes Back), people who are aware of the full story seem to appreciate an apparent 'downer' ending before the triumph of the subsequent episode.

Is it unreasonable to leave a note saying that the story continues in HttT, just as TDG is noted as the continuation of TSG?

Dalas121 commented:
> 'I invoke liberty against servitude!'? Or is that too explicit (in terms of description, not vulgarity)?

Delfador says "servitude" on the very next line, but I can do something similar to this.

> Is it unreasonable to leave a note saying that the story continues in HttT, just as TDG is noted as the continuation of TSG?

Ah, the credits already include _"Story continued in: “Asheviere’s Dogs”"_. (which is then continued by Liberty, and then finished with HttT)

> If the campaign is played on its own, then yes, the ending could seem pretty unsatisfying. But as often with the second part of a trilogy (eg Empire Strikes Back), people who are aware of the full story seem to appreciate an apparent 'downer' ending before the triumph of the subsequent episode.

If TDG ended with a big defeat, I think a downer ending would work fine. But TDG ends with Delfador defeating Eldred and winning a major victory - and I think it'd be strange to cut straight to "_yeah but he lost anyway_".

Wedge009 commented:
> Ah, the credits already include _"Story continued in: “Asheviere’s Dogs”"_. (And AD is continued by Liberty, which is continued by HttT) Does that work?

Oh okay. That should be fine.

> If TDG ended with a big defeat, I think a downer ending would work fine. But TDG ends with Delfador defeating Eldred and winning a major victory - and I think it'd be strange to cut straight to "_yeah but he lost anyway_".

If I read the WML correctly, at present the campaign simply ends after Delfador lands the killing blow, '...and you [Eldred] are dust.' The (presumably) draft epilogue that's disabled-as-comments reads okay to me. ('Queen' may need capitalisation.)

If Deoran was still available, I suppose there could be some discussion between him and Delfador about going into exile, akin to Yoda and Obi-Wan at the end of Revenge of the Sith. I can't remember exactly what sort of character `$companion_id` is supposed to be (guide to the bandits?), but I suppose he/she is not a major enough character to have that epilogue conversation?

Does AD cover Asheviere's consolidation of power, including purging Arand's sons who would otherwise be considered legitimate heirs? I just ask because Konrad is not mentioned at all in this campaign - which is appropriate, I think.

Dalas121 commented:
> If I read the WML correctly, at present the campaign simply ends after Delfador lands the killing blow, '...and you [Eldred] are dust.' The (presumably) draft epilogue that's disabled-as-comments reads okay to me. ('Queen' may need capitalisation.)

Ok, I'll un-comment this and/or do something similar.

> Does AD cover Asheviere's consolidation of power, including purging Arand's sons who would otherwise be considered legitimate heirs? I just ask because Konrad is not mentioned at all in this campaign - which is appropriate, I think.

Yeah, it's mostly about Asheviere asserting her control over the various lords and nobles of Wesnoth. Her killing of Arand's sons is briefly covered in storytext, and Konrad is alluded to but not mentioned by name.

soliton- commented:
This specific case may be obsolete soon but as a quick workaround for wmlindent note that:
```
    {USE_STRONGER_AMLAS (AMLA_FILTER=
        side=1
        [not]
            id=Delfador
        [/not]
    )}
```
can be written as
```
    {USE_STRONGER_AMLAS (
        AMLA_FILTER=
        side=1
        [not]
            id=Delfador
        [/not]
    )}
```
and then wmlindent leaves it that way.

Maybe not as nice but gets you past the check until someone can improve wmlindent.

Dalas121 commented:
Thanks! Yeah, I've still got a lot of stuff I need to fix in this PR

Dalas121 commented:
CodeQL and Map Diff checks are still failing, but I was told that those are OK to ignore.

Any reasons why this shouldn't be merged?

Wedge009 commented:
With regards to the prose, I think you said you've already updated from my review. I can't comment on the CI issues or other technical aspects. From my perspective I think it's okay to merge, especially with TSG having already been updated.

Although, at time of writing I see there are conflicts preventing a merge so perhaps a re-base is needed first - might help with the CI failures too.

Dalas121 commented:
Merging.

Wedge009 commented:
A shame the point release was tagged before this was merged, but maybe it's better that there's only one campaign revision at a time.

Wedge009 commented:
CodeQL check is complaining about being unable to build. Not sure what's going on there.

Wedge009 commented on data/campaigns/The_Deceivers_Gambit/scenarios/07_The_Great_River.cfg:
```diff
@@ -0,0 +1,1014 @@
+        [message]
+            speaker=Ysaxis
+            #po: no matter, my saurains will slip past and warn the army! Everyone, escape to the west!
+            message=_"$part2_message No matter, my sssaurians will ssslip past and warn the army! Everyone, essscape to the west!"
```
@Dalas121 Is this correct? I'm not sure that the variable is supposed to be part of this translation string.

Dalas121 replied:
Fixed, thanks. I've changed it to read `$part2_message+_" No matter, my sssaurians will ssslip past and warn the army! Everyone, essscape to the west!"`

Wedge009 commented on data/campaigns/The_Deceivers_Gambit/scenarios/11_Clan_Blackcrest.cfg:
```diff
@@ -0,0 +1,1017 @@
+            speaker=narrator
+            message= _ "Welcome to the Ford of Abez
+South: Tath
+North:<span strikethrough="true"> Dwarven Doors </span>"
```
@Dalas121 I think this needs to be restructured - the quotes for `"true"` terminates the translation string early.

Dalas121 replied:
Ah, thanks, I'll use single quotes instead.

Dalas121 commented on data/campaigns/The_Deceivers_Gambit/scenarios/13x_The_King_is_Dead.cfg:
```diff
@@ -0,0 +1,232 @@
+            speaker=Delfador
+            message= _ "Scouts, take my seal! Spread the word; all of Wesnoth must know of Eldred’s treachery!
+
+The rest of you, MAKE READY! We march to kill the king."
```
I agree it's a odd, but I'd meant it to deliberately sound a bit dissonant. For some reason I think it sounds better that way; a tacit acknowledgement of Eldred's seizure of power, and the beginning of the protagonists' transition from royal soldiers (TSG, TDG) to rebels (L, HttT).

Wedge009 commented on data/campaigns/The_Deceivers_Gambit/units/Orcish_Sorceress.cfg:
```diff
@@ -0,0 +1,130 @@
+    advances_to=null {AMLA_DEFAULT}
+    cost=54
+    usage=archer
+    description= _ "Orcs are sometimes capable as living as long as humans, but very few individuals ever fully realize that potential, as the average orc only sees two or three decades before meeting their end either in battle. Only the shamans, isolated from the daily strife of the major orcish clans, commonly live to reach old age.
```
'meeting their end either in battle.' ...or?

Dalas121 replied:
Oops, thanks.
